### PR TITLE
Update deps and drop python2

### PR DESCRIPTION
**Pull request by Keats**
## Status
- [ ] **Ready**
- [x] **In progress**

## Description
Drop python2 support
Upgrade all dependencies and see what breaks

### CODE REVIEW

**Comment by jwin4740** [comment]
New to this project. Is there someone I can contact for help onboarding?

File: onecodex/notebooks/exporters.py [hunk 1]
```diff
@@ -70,6 +69,11 @@ def __init__(self, config=None, **kw):
         except ImportError:
             return
 
+    @property
+    def template_paths(self):
+        # https://github.com/jupyter/nbconvert/issues/1492
```
Comment (bcamarda) [inline onecodex/notebooks/exporters.py:74]: should we add this comment before line 172, too?

File: setup.py [hunk 2]
```diff
@@ -38,18 +38,25 @@
     "vega_datasets",
     "black",
```
Comment (jairideout) [inline setup.py:39]: Can you pin `black` to 22.3.0 so that it matches pre-commit?

**Review by jairideout — APPROVED** [review]

File: setup.py (hunk 2 above)
Comment (jairideout) [reply setup.py:39]: Great idea! 👍

**Review by jairideout — APPROVED** [review]

**Pull request merged**